pkg/cache/server/config.go

```diff
@@ -125,17 +143,31 @@ func NewConfig(opts *cacheserveroptions.CompletedOptions) (*Config, error) {
 	serverConfig.LoopbackClientConfig.DisableCompression = true
 	clientutils.EnableMultiCluster(serverConfig.LoopbackClientConfig, &serverConfig.Config, "namespaces", "apiservices", "customresourcedefinitions", "clusterroles", "clusterrolebindings", "roles", "rolebindings", "serviceaccounts", "secrets")
 
+	var err error
+	c.ApiExtensionsClusterClient, err = apiextensionsclient.NewClusterForConfig(serverConfig.LoopbackClientConfig)
```

Comment:
nit: we're moving to the context-based clients instead of `NewCluster` - can we do that here as well to save Varsha some work in the future? :)

Reply:
Look for any use of `logicalcluster.WithCluster()` e.g.:

```go
	return c.kubeClusterClient.CoreV1().Namespaces().Patch(logicalcluster.WithCluster(ctx, clusterName), name, pt, data, opts, subresources...)
```

Reply:
okay, but I need a `Wildcard` logical cluster for informers, it seems like informers don't support the new way of passing a cluster name.

Reply:
Yes, that is the one case where this approach does not work

Reply:
#1813 fixes that :)

Reply:
do we need `2` clients until it merges?

Reply:
Not sure I follow actually - you need a cluster-scoped client for the factory:

```go
# github.com/kcp-dev/kcp/pkg/client/informers
func NewSharedInformerFactoryWithOptions( ... versioned.Interface, ... ) {}
```

Where you can get such a thing from `.Cluster(logicalcluster.Wildcard)` which should be identical to `SetCluster()` on the `*rest.Config`, no?

```go
# github.com/kcp-dev/kcp/pkg/client/clientset/versioned
func (c *Cluster) Cluster(name v2.Name) Interface {}
```


So `NewClusterForConfig(cfg).Cluster(logicalcluster.Wildcard)` would become `NewClientForConfig(kcpclienthelper.SetCluster(rest.CopyConfig(cfg), logicalcluster.Wildcard))`?

Reply:
something is broken, I think I have to use it along with `NewClusterRoundTripper` but then the path seems to be incorrect `/clusters/*/clusters/system:system-crds/apis/apiextensions.k8s.io/v1/customresourcedefinitions/apiresourceschemas.apis.kcp.dev`

I'd like to proceed with this PR if that's okay. I can revisit using the context-based clients in the future